httpx/client.py

```diff
@@ -176,6 +179,13 @@ def merge_cookies(
             response = await self.dispatch.send(
                 request, verify=verify, cert=cert, timeout=timeout
             )
+            # TODO: requests also checks for number of 401 calls < 2
```

Comment:
Instead of reimplementing is_client_error on response, why not use response.status_code.is_client_error?

Reply:
Law of Demeter? Frankly I was simply following the existing `is_redirect`, I'm happy to remove it.